dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -267,6 +318,8 @@ void WebdatasetLoader::PrepareMetadataImpl() {
   }
   copy_read_data_ = dont_use_mmap_ || !mmap_reserver_.CanShareMappedData();
 
+  is_index_generated_ = index_paths_.size() == 0;
```

Comment:
Besides rather poor variable name (see the comment in class definition), I'd suggest to issue the warning here.
```suggestion
  generate_index_ = index_paths_.size() == 0;
  if (generate_index_) {
    DALI_WARN("Index file not provided, it may take some time to infer it from the tar file");
  }
```